src/test/java/net/snowflake/client/jdbc/SnowflakeDriverLatestIT.java

```diff
@@ -1766,6 +1766,93 @@ public void testS3PutInGS() throws Throwable {
+    Properties paramProperties = new Properties();
+    List<String> supportedAccounts = Arrays.asList("gcpaccount", "azureaccount");
+    for (String account : supportedAccounts){
+      try (Connection connection = getConnection(account, paramProperties);
+           Statement statement = connection.createStatement()) {
+        try {
+          // create a stage to put the file in
+          statement.execute("CREATE OR REPLACE STAGE " + testStageName);
+
+          // put compressed and uncompressed file
+          String putCompressedCall =
+                  String.format("PUT file://%s @%s", getFullPathFileInResource(TEST_DATA_FILE), testStageName);
+          String putUncompressedCall =
+                  String.format("PUT file://%s @%s auto_compress=false", getFullPathFileInResource(TEST_DATA_FILE), testStageName);
+          statement.execute(putCompressedCall);
+          statement.execute(putUncompressedCall);
+
+          // get unencrypted contents
+          String getCompressedCall =
+                  String.format(
+                          "select system$it('GET_FILE_FROM_STAGE', '%s', '', '%s', '%s')",
+                          testStageName, TEST_DATA_FILE+".gz", "false");
+          String getUncompressedCall =
+                  String.format(
+                          "select system$it('GET_FILE_FROM_STAGE', '%s', '', '%s', '%s')",
+                          testStageName, TEST_DATA_FILE, "false");
+          ResultSet rs = statement.executeQuery(getCompressedCall);
+          rs.next();
+          String compressedPlaintext = rs.getString(1);
+          rs = statement.executeQuery(getUncompressedCall);
+          rs.next();
+          String uncompressedPlaintext = rs.getString(1);
+
+          // encrypt files
+          String encryptCompressedCall = String.format(
+                  "select system$it('ENCRYPT_STAGE_FILE', '%s', '', '%s')",
+                  testStageName, TEST_DATA_FILE+".gz");
+          String encryptUncompressedCall = String.format(
+                  "select system$it('ENCRYPT_STAGE_FILE', '%s', '', '%s')",
+                  testStageName, TEST_DATA_FILE);
+          try{
+            // compare encrypted with plaintext
+            statement.execute(encryptCompressedCall);
+            statement.execute(encryptUncompressedCall);
+            rs = statement.executeQuery(getCompressedCall);
+            rs.next();
+            String compressedEncrypted = rs.getString(1);
+            rs = statement.executeQuery(getUncompressedCall);
+            rs.next();
+            String uncompressedEncrypted = rs.getString(1);
+            assertNotEquals(compressedPlaintext, compressedEncrypted);
+            assertNotEquals(uncompressedPlaintext, uncompressedEncrypted);
+          }catch (SnowflakeSQLException e){
+            // Azure throws an exception when we try to read it unencrypted. This is expected and means the encryption was successful.
```

Comment:
How about GCP?

Reply:
GCP will return the encrypted text, thus the asserts above comparing to the plaintext.